tests/components/dwd_weather_warnings/test_init.py

```diff
@@ -56,66 +34,67 @@ async def test_load_unload_entry(
+    )
     assert entry.state == ConfigEntryState.SETUP_RETRY
 
 
-async def test_load_missing_device_tracker(hass: HomeAssistant) -> None:
+async def test_load_missing_device_tracker(
+    hass: HomeAssistant, mock_tracker_entry: MockConfigEntry
+) -> None:
     """Test loading the integration with a missing device tracker."""
-    entry = MockConfigEntry(domain=DOMAIN, data=DEMO_TRACKER_CONFIG_ENTRY)
-    entry.add_to_hass(hass)
-
-    await hass.config_entries.async_setup(entry.entry_id)
-    await hass.async_block_till_done()
+    entry = await init_integration(hass, mock_tracker_entry)
     assert entry.state == ConfigEntryState.SETUP_RETRY
 
 
-async def test_load_missing_required_attribute(hass: HomeAssistant) -> None:
+async def test_load_missing_required_attribute(
+    hass: HomeAssistant, mock_tracker_entry: MockConfigEntry
+) -> None:
     """Test loading the integration with a device tracker missing a required attribute."""
-    entry = MockConfigEntry(domain=DOMAIN, data=DEMO_TRACKER_CONFIG_ENTRY)
-    entry.add_to_hass(hass)
-
+    mock_tracker_entry.add_to_hass(hass)
     hass.states.async_set(
-        DEMO_TRACKER_CONFIG_ENTRY[CONF_REGION_DEVICE_TRACKER],
+        mock_tracker_entry.data[CONF_REGION_DEVICE_TRACKER],
         STATE_HOME,
         {ATTR_LONGITUDE: "7.610263"},
     )
 
-    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.config_entries.async_setup(mock_tracker_entry.entry_id)
     await hass.async_block_till_done()
-    assert entry.state == ConfigEntryState.SETUP_RETRY
+    assert mock_tracker_entry.state == ConfigEntryState.SETUP_RETRY
 
 
 async def test_load_valid_device_tracker(
-    hass: HomeAssistant, entity_registry: er.EntityRegistry, mock_dwdwfsapi: MagicMock
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    mock_tracker_entry: MockConfigEntry,
+    mock_dwdwfsapi: MagicMock,
 ) -> None:
     """Test loading the integration with a valid device tracker based entry."""
-    entry = MockConfigEntry(domain=DOMAIN, data=DEMO_TRACKER_CONFIG_ENTRY)
-    entry.add_to_hass(hass)
+    mock_tracker_entry.add_to_hass(hass)
     entity_registry.async_get_or_create(
         "device_tracker",
-        entry.domain,
+        mock_tracker_entry.domain,
         "uuid",
         suggested_object_id="test_gps",
-        config_entry=entry,
+        config_entry=mock_tracker_entry,
     )
 
     hass.states.async_set(
-        DEMO_TRACKER_CONFIG_ENTRY[CONF_REGION_DEVICE_TRACKER],
+        mock_tracker_entry.data[CONF_REGION_DEVICE_TRACKER],
         STATE_HOME,
         {ATTR_LATITUDE: "50.180454", ATTR_LONGITUDE: "7.610263"},
     )
 
-    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.config_entries.async_setup(mock_tracker_entry.entry_id)
     await hass.async_block_till_done()
 
-    assert entry.state == ConfigEntryState.LOADED
-    assert entry.entry_id in hass.data[DOMAIN]
+    assert mock_tracker_entry.state == ConfigEntryState.LOADED
```

Comment:
is

Reply:
Updated in e3a1f118391d1cdc95dbd0e1e5e0feac21fb472d